ddtrace/tracer/baggage.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package tracer
+
+import (
+	"context"
+)
+
+// baggageKey is an unexported type used as a context key. It is used to store baggage in the context.
+// We use a struct{} so it won't conflict with keys from other packages.
+type baggageKey struct{}
+
+// baggageMap retrieves the map that holds baggage from the context.
+// Returns nil if no baggage map is stored in the context.
+func baggageMap(ctx context.Context) map[string]string {
+	bm, _ := ctx.Value(baggageKey{}).(map[string]string)
```

Comment:
I believe this could cause a panic when converting to a map type, if ctx.Value(baggageKey{}) is nil. You can separate the check from the type conversion, making it two steps instead of one.